src/components/widget/MultiSelect.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class MultiSelect extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: this.props.options,
+      filter: '',
+      checkedItems: {},
+    };
+  }
+
+  componentDidMount() {
+    this.props.onFocus();
+  }
+
+  static getDerivedStateFromProps(nextProps) {
+    let selected =
+      nextProps.selectedItems && Array.isArray(nextProps.selectedItems)
+        ? nextProps.selectedItems
+        : nextProps.selectedItems !== null
+        ? nextProps.selectedItems.values
+        : null;
+    if (selected !== null) {
+      let updatedCheckedItems = {};
+      selected.map((item) => {
+        if (item) {
+          updatedCheckedItems[item.key] = {
+            key: item.key,
+            caption: item.caption,
+            value: true,
+          };
+        }
+        return item;
+      });
+      return {
+        checkedItems: updatedCheckedItems,
+      };
+    }
+
+    return null;
+  }
+
+  selectItem = (key, caption) => {
+    let selected = null;
+    let newCheckedItems = JSON.parse(JSON.stringify(this.state.checkedItems));
+    if (typeof this.state.checkedItems[key] === 'undefined') {
```

Comment:
Maybe we could do something like :
```javascript
_.set(checkedItems, ['key1'], { key: 'key1', value: !_.get(checkedItems, ['key1', 'value'], true)});
```

Reply:
No action should be done to directly alter the state. That is the reason I deep cloned the newCheckedItems in the first place.

Reply:
Also didn't want to . add lodash and preferred the vanillaJS way :). Thanks for understanding .